emlearn/eml_mixture.h

```diff
@@ -287,6 +287,31 @@ eml_mixture_score(EmlMixtureModel *model,
     return EmlOk;
 }
 
+int32_t eml_mixture_predict(EmlMixtureModel *model,
+                    const float values[], int32_t values_length,
+                    float *probabilities,
+                    float *out_score, float *out_resp) {
+    EML_PRECONDITION(model, -EmlUninitialized);
+    EML_PRECONDITION(values, -EmlUninitialized);
+    EML_PRECONDITION(model->n_components > 0, -EmlUninitialized);
+
+    EmlError status = \
+        eml_mixture_score(model, values, values_length, probabilities, out_score);
+    if (status != EmlOk) {
+        return status;
+    }
+    float* log_resp = (float*)malloc(model -> n_components * sizeof(float)); // defined an array log responsibilities 'dynamically
```

Comment:
Dynamic allocation is not allowed in emlearn. And this one even leaks memory forever (no free)... Avoid creating a new buffer, for example, by reuse one of the existing buffers that are passed in. It seems here that the log_resp temporary is not needed at all?

Reply:
Done ✅ Thanks for the feedback! I blundered a memory leak!